### PR TITLE
Enable stress jobs on x86 ryujit

### PR DESCRIPTION
**Pull request by adiaaida**
We want to be able to run jitstress, gcstress, etc. jobs on ryujit to
monitor our progress. This change adds additional vars to be written to
the TestEnv script for jitstress jobs.

### CODE REVIEW

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

File: netci.groovy [hunk 1]
```diff
@@ -1111,6 +1111,195 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
...
+                        Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${arch} ${jit} ${configuration} Long-Running GC Build & Test", "(?i).*test\\W+${os}\\W+${arch}\\W+${jit}\\W+${configuration}\\W+${scenario}.*")
+                    }
+                    break
+                case 'gcsimulator':
+                    if (configuration == 'Release') {
+                        Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${arch} ${jit} ${configuration} GC Simulator", "(?i).*test\\W+${os}\\W+${arch}\\W+${jit}\\W+${configuration}\\W+${scenario}.*")
+                    }
+                    break
+                case 'minopts':
+                    assert (os == 'Windows_NT') || (os in Constants.crossList)
```
Comment (jashook) [inline netci.groovy]: We should not generate x86 jobs for linux. This assert should assert Windows_NT

File: netci.groovy [hunk 2]
```diff
@@ -1111,6 +1111,195 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
...
+                case 'gcstress0xc_zapdisable':
+                case 'gcstress0xc_zapdisable_jitstress2':
+                case 'gcstress0xc_zapdisable_heapverify1':
+                case 'gcstress0xc_jitstress1':
+                case 'gcstress0xc_jitstress2':
+                case 'gcstress0xc_minopts_heapverify1':                                 
+                    def displayStr = getStressModeDisplayName(scenario)
+                    assert (os == 'Windows_NT') || (os in Constants.crossList)
+                    Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${arch} ${jit} ${configuration} Build and Test (Jit - ${displayStr})",
+                       "(?i).*test\\W+${os}\\W+${arch}\\W+${jit}\\W+${configuration}\\W+${scenario}.*")
```
Comment (jashook) [inline netci.groovy]: Could you add all of these PR triggers to the documentation?

Comment (adiaaida) [inline netci.groovy]: This is a WIP. I'm really just trying to see if it works right now.

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

File: netci.groovy (hunk 2 above)
Comment (jashook) [inline netci.groovy]: Ya I gotcha.

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@RussKeldorph can you check the jit stress jobs list in this PR and make sure I've included all the tests we are interested in. Listing (assuming it doesn't get overwritten): http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/GenPRTest/job/jitstress/

File: netci.groovy (hunk 1 above)
Comment (adiaaida) [inline netci.groovy]: Done

File: netci.groovy (hunk 2 above)
Comment (adiaaida) [inline netci.groovy]: Done.

**Comment by jashook** [comment]
lgtm

**Comment by RussKeldorph** [comment]
@adiaaida Looks right.  

@sivarv @CarolEidt I'm assuming since RyuJIT/x86 is using LSRA then the same JitStressRegs modes are valid.  If that's not true, let us know.  Acknowledged we aren't currently running all JitStressRegs modes in CI.  The current set was based on prior sign-off crit. and may need to be reevaluated separate from this change.

LGTM

**Comment by adiaaida** [comment]
@mmitche PTAL

File: netci.groovy [hunk 3]
```diff
@@ -1111,6 +1111,195 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
...
+                case 'default':
+                    // Default trigger
+                    if (os == 'Windows_NT') {
+                        if (configuration == 'Checked') {
+                            Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${arch} ${jit} ${configuration} Build and Test")
+                        }
+                    }
+                    else {
+                        // default trigger
+                        Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${arch} ${jit} ${configuration} Build", "(?i).*test\\W+${arch}\\W+${osGroup}.\\W+${jit}*")
```
Comment (mmitche) [inline netci.groovy]: Should be .* at the end

**Comment by mmitche** [comment]
LGTM other than minor nit.

File: netci.groovy (hunk 3 above)
Comment (adiaaida) [inline netci.groovy]: Huh. That was a copy-paste. I'll fix it where I copied from too.

**Comment by adiaaida** [comment]
This closes #6924.

**Pull request merged**